client/state/posts/selectors.js

```diff
@@ -359,6 +360,29 @@ export const isEditedPostDirty = createSelector(
 );
 
 /**
+  * Returns true if the edited post has content
+  * (title, excerpt or content not empty)
+  *
+  * @param  {Object}  state  Global state tree
+  * @param  {Number}  siteId Site ID
+  * @param  {Number}  postId Post ID
+  * @return {Boolean}        Whether the edited post has content or not
+ */
+export const editedPostHasContent = createSelector(
+	( state, siteId, postId ) => {
+		const editedPost = getEditedPost( state, siteId, postId );
+		return (
+			!! editedPost &&
+			(
+				some( [ 'title', 'excerpt' ], ( field ) => editedPost[ field ] && !! editedPost[ field ].trim() ) ||
+				! isPostContentEmpty( editedPost.content )
+			)
+		);
+	},
+	( state ) => [ state.posts.items, state.posts.edits ]
+);
```

Comment:
> I don't have the experience yet with the editor to justify that it's really needed here

nor have I the experience to justify nixing it! :wink:

> I did a small performance benchmarking

thanks! this is helpful. personally I would say that `0.3ms` is very near zero but large in terms of actual processor usage. how often does this get called? on every keystroke in the editor? on every dispatched action passed through the reducers?

also, those timings look suspicious (especially `0ms`) - did you run them with `Performance.now()` by any chance instead of `Date.now()`? (I guess I should just ask, how did you measure it?)

> Is it worth it?

that probably hearkens back to how often it happens and more precisely what the difference is. I certainly don't want to push out something that works against the awesome performance work that has been going on. if we leave this in here I think a comment is warranted

**update**
ooh ooh! 🖐 why is not `getEditedPost()` the selector which is memoized? wouldn't it make far more sense there since we are saying that _it's_ the one that's slow, not this one or presumably any of the other twenty-some-odd number of things that call it?